aws/resource_aws_iam_service_linked_role_test.go

```diff
@@ -0,0 +1,90 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAwsIamServiceLinkedRole(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsIamServiceLinkedRoleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(testAccAwsIamServiceLinkedRoleConfig, "elasticbeanstalk.amazonaws.com"),
+				Check:  testAccCheckAwsIamServiceLinkedRoleExists("aws_iam_service_linked_role.test"),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsIamServiceLinkedRoleDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).iamconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_iam_service_linked_role" {
+			continue
+		}
+
+		service_name := rs.Primary.ID
```

Comment:
When we update `d.SetId()` this will be the role ARN and we can parse for the role name to call `GetRole()` 👍